### PR TITLE
[UWP] Prevents Picker collapsing on opening Dropdown menu

### LINKED ISSUE #6403: Xamarin.Forms UWP Picker collapses on opening Dropdown menu [Bug]
picker_uwp
In UWP Apps the Picker control does collapse the entire control when opening the Dropdown menu. This results in a jumping layout, especially if any other controls in same flexible layout hierarchy. 
Important side note: The title must be set and should be smaller than the size of the picker item strings. Also the picker color property must be set (which anyway has a default value).

### Steps to Reproduce
1.  Use a Picker control, define 2 or more string items, do NOT set WidthRequest or MinimumWidthRequest
2.  Run the App, open the Picker Dropdown menu 
3.  --> entire Pickers (behind the Dropdown menu) collapses.
4. After closing or selecting a item the Pickers gets visible again.

### Expected Behavior
The Picker should stay visible with its actual width and height until a other item is selected or the Dropdown menu is otherwise closed somehow.

- Version with issue: Xamarin.Forms UWP, 4.0.0.471375 and 4.0.0.482894
- IDE: VS2017

Workaround: Set a WidthRequest and/or MinimumWidthRequest.

Sample Code:
```
                    <FlexLayout Direction="Row" Wrap="Wrap" JustifyContent="Start" AlignItems="Start" AlignContent="Start">
                        <Picker VerticalOptions="Center" Margin="0" 
                                TextColor="Blue" Title="Test"
                                SelectedIndex="2" SelectedItem="{Binding SelectedDevice}">
                            <Picker.Items>
                                <x:String>Device A</x:String>
                                <x:String>Custom Device B</x:String>
                                <x:String>Latest Prototype C</x:String>
                            </Picker.Items>
                        </Picker>
                        <Entry Placeholder="Serial no" VerticalOptions="Center" Text="Serial" FontSize="16" WidthRequest="100" MinimumWidthRequest="100" MaxLength="6" Keyboard="Text"/>
                        <Button Text="Add device" VerticalOptions="Center" FontSize="16" Padding="0" BorderWidth="0"/>
                    </FlexLayout>
```

### PR DESCRIPTION
**Pull request by bmacombe**
# Description of Change ###
Changes the behavior of the Picker on UWP by preventing the UWP Combobox from altering it's width when the drop down opens.

I considered trying to retain the UWP MinWidth of the control, but it doesn't seem to be used.

### Issues Resolved ### 

- fixes #6403 

### API Changes ###
 None

### Platforms Affected ### 

- UWP

### Behavioral/Visual Changes ###
Prevents the UWP combo box from shrinking to min size when the drop down opens and a width is not set.

### Before/After Screenshots ### 

Before
58880478-40d0e280-86d8-11e9-9163-e31f52f656d3

After
After

### Testing Procedure ###
Open the issue page and select various items from the picker.  Ensure that the label to the left does not move on the drop down open, only after selecting an item.

### PR Checklist ###

- [x] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue6403.xaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+<controls:TestContentPage xmlns="http://xamarin.com/schemas/2014/forms"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:d="http://xamarin.com/schemas/2014/forms/design"
+             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+             xmlns:controls="clr-namespace:Xamarin.Forms.Controls"
+             mc:Ignorable="d"
+             x:Class="Xamarin.Forms.Controls.Issues.Issue6403">
+    <StackLayout>
+
+        <Label Text="Select different items from the picker.  It's width should only change after selection."></Label>
```
Comment (samhouts) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue6403.xaml]:
```suggestion
        <Label Text="Select different items from the picker.  Its width should only change after selection."></Label>
```

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue6403.xaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        <FlexLayout Direction="Row" >
+
+            <!-- issue also occurs if Title is empty, but than different behaviour (max size is dropdown menu size) -->
```
Comment (samhouts) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue6403.xaml]:
```suggestion
            <!-- issue also occurs if Title is empty, but then different behaviour (max size is dropdown menu size) -->
```

**Comment by samhouts** [comment]
@bmacombe Looks like there might have been something bad about the merge. Build is failing.

**Comment by bmacombe** [comment]
@samhouts Thanks for letting me know, I'll take a look

**Comment by bmacombe** [comment]
@samhouts I believe this issue was some prior commit also added a ctor.  I merged them together and all builds locally.

**Comment by bmacombe** [comment]
@samhouts Would you kick off a build when you have a chance?  I just want to make sure this all good.

Thanks!

**Comment by samhouts** [comment]
It's off! 🤞

**Comment by bmacombe** [comment]
@samhouts  Thanks...looks like the ctor was ok, but there is something else.  I'll take another look.

**Comment by bmacombe** [comment]
@samhouts I just went back to the last commit before the merge of 4.5, then rebased on 4.5 and cherry picked the last few commits into it.

Mind starting another build?  Thanks!

**Comment by bmacombe** [comment]
@samhouts I did a rebuild vs just starting UWP control gallery and some of the Android projects are failing local.  I'm not sure what is up right now because I don't think I made any changes that should affect that.

File: Xamarin.Forms.Platform.UAP/FormsComboBox.cs [hunk 3]
```diff
@@ -7,17 +7,35 @@ namespace Xamarin.Forms.Platform.UWP
...
+		}
 
-		internal bool IsOpeningAnimated { get; private set; }
+		void FormsComboBox_DropDownOpened(object sender, object e)
+		{
+			MinWidth = ActualWidth;
+		}
 
-		public FormsComboBox() 
+		void FormsComboBox_SelectionChanged(object sender, Windows.UI.Xaml.Controls.SelectionChangedEventArgs e)
```
Comment (rmarinho) [inline Xamarin.Forms.Platform.UAP/FormsComboBox.cs]:
```suggestion
		void FormsComboBoxSelectionChanged(object sender, Windows.UI.Xaml.Controls.SelectionChangedEventArgs e)
```

File: Xamarin.Forms.Platform.UAP/FormsComboBox.cs [hunk 4]
```diff
@@ -7,17 +7,35 @@ namespace Xamarin.Forms.Platform.UWP
...
 
-		internal bool IsFullScreen => Device.Idiom == TargetIdiom.Phone && Items != null && Items.Count > 5;
+		~FormsComboBox()
+		{
+			DropDownOpened -= FormsComboBox_DropDownOpened;
+			SelectionChanged -= FormsComboBox_SelectionChanged;
+		}
 
-		internal bool IsOpeningAnimated { get; private set; }
+		void FormsComboBox_DropDownOpened(object sender, object e)
```
Comment (rmarinho) [inline Xamarin.Forms.Platform.UAP/FormsComboBox.cs]:
```suggestion
		void FormsComboBoxDropDownOpened(object sender, object e)
```

**Comment by bmacombe** [comment]
This branch is still giving me fits on building...I'm going to try to pull on my changes out again and start a fresh one @samhouts what branch should I target this at? 4.5 still?

File: Xamarin.Forms.Platform.UAP/FormsComboBox.cs [hunk 5]
```diff
@@ -7,17 +7,35 @@ namespace Xamarin.Forms.Platform.UWP
...
+		{
+			DefaultStyleKey = typeof(FormsComboBox);
+			DropDownOpened += FormsComboBoxDropDownOpened;
+			SelectionChanged += FormsComboBoxSelectionChanged;
+		}
 
-		internal bool IsFullScreen => Device.Idiom == TargetIdiom.Phone && Items != null && Items.Count > 5;
+		~FormsComboBox()
+		{
+			DropDownOpened -= FormsComboBoxDropDownOpened;
```
Comment (PureWeen) [inline Xamarin.Forms.Platform.UAP/FormsComboBox.cs]: These aren't needed

**Comment by PureWeen** [comment]
@bmacombe what exceptions are you getting?

File: Xamarin.Forms.Platform.UAP/FormsComboBox.cs (hunk 5 above)
Comment (bmacombe) [reply Xamarin.Forms.Platform.UAP/FormsComboBox.cs]: I wondered about that...I'll remove them

**Comment by bmacombe** [comment]
@PureWeen I'll run some more builds now and get you an idea...you could kick off a CI build also, it happened there beside my local build.  It's strange, I ran a build of the UWP control gallery after renaming the methods tonight and it was fine.  Then I did a clean/rebuild to make sure to build everything else and it had a number of what looked like Android resource errors.

There was a conflict when Sam merged in the 4.5 branch, but I went back to a commit before that, rebased on 4.5 and then cherry picked the other changes and still had issues.

**Comment by bmacombe** [comment]
@PureWeen Here are some examples from my local build just now
image

**Comment by PureWeen** [comment]
@bmacombe usually I can fix those by either

- right click rebuild Android Control Gallery

or 

- git clean -xdf
- reopen sln

**Comment by bmacombe** [comment]
@PureWeen Thanks, I always forget about the git clean -xdf.  I got it building...then rebased on 4.5 latest for proj conflicts, screwed up a merge in the rebase, started over on a new rebase.

But I think it all builds locally now! (just no cooperating mac, so no iOS)  Would you or @samhouts start a CI build? Curious to see the problems from the previous CI builds are fixed.

**Comment by bmacombe** [comment]
@PureWeen Looks like the CI build failures are doc related issues in the nuget generation.

Is that anything I need to look at?

**Comment by PureWeen** [comment]
@bmacombe that's all us :-/ 

I'll let you know once those are fixed and then you'll most likely need to rebase

**Comment by PureWeen** [comment]
alright @bmacombe

Try rebasing and then we'll kick this off again!

**Comment by bmacombe** [comment]
@PureWeen rebased and local build good!

**Pull request merged**